src/Detector/PefDetector.php

```diff
@@ -27,8 +27,8 @@ public function detect(SplFileObject $file): ?ImageType
             return null;
         }
 
-        $bytes = $file->fread(2);
+        $bytes = $file->fread(510);
 
-        return $bytes === "\0*" ? new ImageType(ImageType::PEF) : null;
+        return strpos($bytes ?: '', "PENTAX") >= 6 ? new ImageType(ImageType::PEF) : null;
```

Comment:
- It cannot only detect the bytes are `\0*` to identify the image is `PEF` because the `HDR` image have the same bytes in the same position.
- The `PENTAX` bytes only present in the `PEF` image so we should detect these `bytes` to make sure the image is `PEF` format correctly.